### PR TITLE
Remove deprecated `TestTubeLogger`

### LINKED ISSUE #12858: Remove deprecated `TestTubeLogger`
## What does this PR do?

Part of #12521 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Yes!

Make sure you had fun coding 🙃

### PR DESCRIPTION
**Pull request by stanbiryukov**
## What does this PR do?

Fixes #12858, which is subset of #12521

### Does your PR introduce any breaking changes? If yes, please list them.

Removes `from pytorch_lightning.loggers import TestTubeLogger`

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Yes!
Make sure you had fun coding 🙃

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -61,7 +61,6 @@ module = [
     "pytorch_lightning.loggers.mlflow",
     "pytorch_lightning.loggers.neptune",
     "pytorch_lightning.loggers.tensorboard",
-    "pytorch_lightning.loggers.test_tube",
```
Comment (akihironitta) [inline pyproject.toml:64]: Nice catch! :)

**Review by akihironitta — COMMENTED** [review]
Hi @stanbiryukov Thank you for your work! The CI jobs are failing due to the import statement here. Would you mind removing this import statement, too?
```
  File "/home/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/loggers/__init__.py", line 28, in <module>
    from pytorch_lightning.loggers.test_tube import _TESTTUBE_AVAILABLE, TestTubeLogger  # noqa: F401
ModuleNotFoundError: No module named 'pytorch_lightning.loggers.test_tube'
```

**Comment by stanbiryukov** [comment]
> Hi @stanbiryukov Thank you for your work! The CI jobs are failing due to the import statement here. Would you mind removing this import statement, too?
> 
> ```
>   File "/home/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/loggers/__init__.py", line 28, in <module>
>     from pytorch_lightning.loggers.test_tube import _TESTTUBE_AVAILABLE, TestTubeLogger  # noqa: F401
> ModuleNotFoundError: No module named 'pytorch_lightning.loggers.test_tube'
> ```

Thanks, @akihironitta. Good point on the `__init__.py` file. I also updated `tests/loggers/test_all.py` so those now pass without `test_tube`.

**Review by akihironitta — COMMENTED** [review]
Once an entry is added to the changelog, LGTM!

**Comment by stanbiryukov** [comment]
> Once an entry is added to the changelog, LGTM!

Thanks @akihironitta - added the removal entry to CHANGELOG.md

**Review by akihironitta — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**